### PR TITLE
[stable/postgresql] Add dedicated tmpfs for /dev/shm.

### PR DESCRIPTION
**Pull request by desaintmartin**
#### What this PR does / why we need it:
Start a database pod without limitations on shm memory. By default docker
limit  to 64MB (see e.g. the [docker
issue](https://github.com/docker-library/postgres/issues/416), which could be
not enough if PostgreSQL uses parallel workers heavily. If this option is
present and value is , to the target database pod will be mounted a new
tmpfs volume to remove this limitation.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/postgresql/templates/statefulset-slaves.yaml [hunk 1]
```diff
@@ -83,12 +83,19 @@ spec:
               chmod 700 {{ .Values.persistence.mountPath }}/data
               find {{ .Values.persistence.mountPath }} -mindepth 0 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | \
                 xargs chown -R {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }}
+{{- if .Values.shmVolume.enable }}
```
Comment (desaintmartin) [inline stable/postgresql/templates/statefulset-slaves.yaml]: What indentation would you like there?

File: stable/postgresql/README.md [hunk 2]
```diff
@@ -116,6 +116,7 @@ The following tables lists the configurable parameters of the PostgreSQL chart a
 | `service.loadBalancerIP`                      | loadBalancerIP if service type is `LoadBalancer`                                                                                                                          | `nil`                                                         |
 | `service.loadBalancerSourceRanges`            | Address that are allowed when svc is LoadBalancer                                                                                                                         | []                                                            |
 | `schedulerName`                               | Name of the k8s scheduler (other than default)                                                                                                                            | `nil`                                                         |
+| `enableShmVolume`                             | Enable emptyDir volume for /dev/shm                                                                                                                                       | `true`                                                        |
```
Comment (desaintmartin) [inline stable/postgresql/README.md]: Is it okay for you to have, like persistence, one global value to enable for both master and slaves?

Comment (juan131) [reply stable/postgresql/README.md]: Yes, but please add that information to the Description:
```
Enable emptyDir volume for /dev/shm both on master and slaves replicas
```

File: stable/postgresql/templates/statefulset-slaves.yaml (hunk 1 above)
Comment (juan131) [reply stable/postgresql/templates/statefulset-slaves.yaml]: I would align it with the `chmod` command

File: stable/postgresql/templates/statefulset-slaves.yaml [hunk 3]
```diff
@@ -83,12 +83,19 @@ spec:
...
           securityContext:
             runAsUser: {{ .Values.volumePermissions.securityContext.runAsUser }}
           volumeMounts:
             - name: data
               mountPath: {{ .Values.persistence.mountPath }}
               subPath: {{ .Values.persistence.subPath }}
+{{- if .Values.shmVolume.enable }}
+            - name: dshm
+              mountPath: /dev/shm
+{{- end }}
```
Comment (juan131) [inline stable/postgresql/templates/statefulset-slaves.yaml]: Could you indent the if/else block?

Comment (juan131) [reply stable/postgresql/templates/statefulset-slaves.yaml]: Apply it to the rest of the manifest

**Review by juan131 — COMMENTED** [review]
Hi @desaintmartin 

Thanks so much for this contribution! I quick question:

- Are containers affected by this limitation when using [containerd](https://containerd.io/) as runtime too?

**Comment by desaintmartin** [comment]
> Hi @desaintmartin
> 
> Thanks so much for this contribution! I quick question:
> 
>     * Are containers affected by this limitation when using [containerd](https://containerd.io/) as runtime too?

Hm, this is not clear: https://github.com/containerd/containerd/issues/3654 states that the default is 64MB (with source as reference) and https://github.com/containerd/containerd/commit/20d3fae3dbc9aa07b1a18c7159f1a8df41347d75 solves it but I can't see this function called anywhere in containerd repo. 
So I would say that by default, it is still 64MB.

**Comment by desaintmartin** [comment]
Actually, I realized that if persistence is not enabled, this is will run. I'll do another initContainer.

**Review by juan131 — APPROVED** [review]
Thanks @desaintmartin !

Let's merge this and wait for docker/containerd to provide a better solution

/lgtm

**Pull request merged**